src/solutions/basic_solutions.jl

```diff
@@ -77,3 +77,15 @@ function build_solution(prob::AbstractIntegralProblem,
                      typeof(prob), typeof(alg), typeof(chi)}(u, resid, prob, alg, retcode,
                                                              chi)
 end
+
+function wrap_sol(sol)
+    sys_type = if sol.prob.f.sys !== nothing
+        sol.prob.f.sys.metadata
```

Comment:
shouldn't this just be done via `prob.problem_type` information?

Reply:
See https://github.com/SciML/DiffEqBase.jl/pull/825

Reply:
why wait to change that?

Reply:
Done